plugin/src/main/java/org/opensearch/ml/task/MLTaskManager.java

```diff
@@ -541,32 +541,81 @@ private ActionListener<UpdateResponse> getUpdateResponseListener(String taskId,
+                    lockDurationSeconds,
+                    null,
+                    MLJobType.BATCH_TASK_UPDATE
+                );
+                IndexRequest indexRequest = new IndexRequest()
+                    .index(CommonValue.ML_JOBS_INDEX)
+                    .id(id)
+                    .source(jobParameter.toXContent(JsonXContent.contentBuilder(), null))
+                    .setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE);
+
+                try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
+                    client.index(indexRequest, ActionListener.runBefore(ActionListener.wrap(r -> {
+                        log.info("Indexed ml task polling job successfully");
+                        this.taskPollingJobStarted = true;
+                    }, e -> log.error("Failed to index task polling job", e)), context::restore));
+                }
+            }
+        }, e -> log.error("Failed to initialize ML jobs index", e)));
+    }
+
+    public void startStatsCollectorJob() {
+        mlIndicesHandler.initMLJobsIndex(ActionListener.wrap(success -> {
+            if (success) {
+                try {
+                    int intervalInMinutes = 5;
+                    Long lockDurationSeconds = 60L;
+
+                    MLJobParameter jobParameter = new MLJobParameter(
+                        MLJobType.STATS_COLLECTOR.name(),
+                        new IntervalSchedule(Instant.now(), intervalInMinutes, ChronoUnit.MINUTES),
+                        lockDurationSeconds,
+                        null,
+                        MLJobType.STATS_COLLECTOR
+                    );
+
+                    IndexRequest indexRequest = new IndexRequest()
+                        .index(CommonValue.ML_JOBS_INDEX)
+                        .id(MLJobType.STATS_COLLECTOR.name())
+                        .source(jobParameter.toXContent(JsonXContent.contentBuilder(), null))
+                        .setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE);
+
+                    try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
+                        client
+                            .index(
+                                indexRequest,
+                                ActionListener
+                                    .runBefore(
+                                        ActionListener
+                                            .wrap(
+                                                r -> log.info("Indexed ml stats collection job successfully"),
+                                                e -> log.error("Failed to index stats collection job", e)
+                                            ),
+                                        context::restore
+                                    )
+                            );
+                    }
+                } catch (IOException e) {
```

Comment:
Why this IOException is not caught in the first method? I see they are almost identical.